src/main/java/org/spongepowered/common/mixin/entitycollisions/ChunkMixin_Collisions.java

```diff
@@ -124,4 +128,18 @@
 
         return true;
     }
+
+    /**
+     * Removes the limit when the bounding box is large. Fixes bugs like
+     * https://github.com/SpongePowered/SpongeCommon/issues/2452.
+     *
+     * @author JBYoshi
+     */
+    private boolean collisionsImpl$isLimitExempt(AxisAlignedBB aabb) {
+        if (!(this.world instanceof WorldServer)) {
```

Comment:
If we are keeping this logic, I'd much rather this be checking that `((WorldBridge) this.world).bridge$isFake()`